crypto/ec/ec_key.c

```diff
@@ -119,9 +119,11 @@ EC_KEY *EC_KEY_copy(EC_KEY *dest, const EC_KEY *src)
                     return NULL;
             }
             if (!BN_copy(dest->priv_key, src->priv_key))
+                BN_free(dest->priv_key);
```

Comment:
Any memory that function allocates must be freed by that function up until the point that it is assigned to one of the EC_KEY objects. At that point ownership is passed to the EC_KEY, and it will get freed when the calling application frees the EC_KEY. In this case the memory is allocated and ownership is transferred to the EC_KEY (i.e. `dest` in this case) in a single step:
````
                dest->priv_key = BN_new();
````
If it had stored the return of `BN_new()` in some local variable without assigning it to `dest->priv_key` then the function would have needed to free it before returning.